script.js

```diff
@@ -1,4 +1,193 @@
+    }
+  }
+};
+// first message that the players see - it's the same for both p1 and p2.
+var firstMessage = function (dice1, dice2, playerNumberFirstMsg) {
+  return `Hello Player ${playerNumberFirstMsg}. <br><br> Your first roll is ${dice1}. <br> Your second roll is ${dice2}. <br>If you want the first roll be in front, just press Submit. <br> If you want the second roll to be first, enter 2. <br><br> Your current number is ${dice1}${dice2}`;
+};
+//second message that the players see - it's the same for both p1 and p2
+var secondMessage = function (playerNumberSecondMsg) {
+  if (playerNumberSecondMsg == 1) {
+    return `Hello Player ${playerNumberSecondMsg}. <br> Your final number is ${player1Array[0]}${player1Array[1]}. It's Player 2's turn now!`;
+  } else if (playerNumberSecondMsg == 2) {
+    return `Hello Player ${playerNumberSecondMsg}. <br> Your final number is ${player2Array[0]}${player2Array[1]}. It's Player 1's turn now!`;
+  }
+};
+//first step of the game where both dice are rolled
+var gameFirstStep = function (mainDice1, mainDice2, playerNumberFirstStep) {
+  // default message once player x presses submit
+  var defaultMessage = firstMessage(
+    mainDice1,
+    mainDice2,
+    playerNumberFirstStep
+  );
+  return defaultMessage;
+};
+//second step of the game where the player gets to input which value they wants to be first
+var gameSecondStep = function (
+  playerInput,
+  playerNumberSecondStep,
+  diceInsert1,
+  diceInsert2
+) {
+  //game step 2 is when the player 1 has said which dice they wanna put first
+  insertArray(playerInput, diceInsert1, diceInsert2, playerNumberSecondStep);
+  var defaultMessage = secondMessage(playerNumberSecondStep);
+  return defaultMessage;
+};
+// kinda useless, i wanted to create a function that would make the main function as clean as possible
+var playerGameSequence = function (
+  diceGameSeq1,
+  diceGameSeq2,
+  dicePos,
+  playerNumberGameSeq
+) {
+  gameFirstStep(diceGameSeq1, diceGameSeq2, playerNumberGameSeq);
+  gameSecondStep(dicePos, playerNumberGameSeq);
+};
+//compare who's tens are bigger. return player value if their tens are bigger
+var isTensBigger = function () {
+  if (player1Array[0] > player2Array[0]) {
+    return 1;
+  } else if (player1Array[0] == player2Array[0]) {
+    return 0;
+  } else {
+    return 2;
+  }
+};
+//compare who's ones are bigger. return player value if their ones are bigger
+var isOnesBigger = function () {
+  if (player1Array[1] > player2Array[1]) {
+    return 1;
+  } else if (player1Array[1] == player2Array[1]) {
+    return 0;
+  } else {
+    return 2;
+  }
+};
+// to see which player has the bigger number
+var winCondition = function () {
+  // if p1 tens is equal to p2 tens
+  if (isTensBigger() == 0) {
+    //return isOnesBigger. We can deterine the winner/draw from isOnesBigger
+    return isOnesBigger();
+  } else {
+    //if tens is not equal, isTensBigger will return the winning player.
+    return isTensBigger();
+  }
+};
```

Comment:
Nice! First time seeing it solved this way. Find the use of functions in functions quite interesting, you're getting a better understanding of it

commonly the numbers in the array will be made into a string that is joined together eg. "5"+"2"
and converted back into a number to be compared.